### PR TITLE
Power Management: Namespace cleanups

### PR DESCRIPTION
**Pull request by nashif**
- Kconfig unification
- API changes to align on one prefix
- Various other minor changes and cleanups

### CODE REVIEW

File: arch/arc/core/reset.S [hunk 1]
```diff
@@ -121,6 +121,7 @@ invalidate_dcache:
 
 done_cache_invalidate:
 
+<<<<<<< HEAD
```
Comment (abrodkin) [inline arch/arc/core/reset.S:124]: Left-over of some work? :)

File: doc/releases/release-notes-2.3.rst [hunk 2]
```diff
@@ -838,7 +838,7 @@ release:
 * :github:`25369` - tests/drivers/gpio/gpio_basic_api: test_gpio_deprecated step fails on STM32 boards
 * :github:`25366` - tests/drivers/counter/counter_basic_api: instable test status on STM32 boards
 * :github:`25363` - tests/drivers/counter/counter_basic_api: Assertion failed on STM32 boards
-* :github:`25354` - Fails to compile when SYS_PM_DIRECT_FORCE_MODE is true
+* :github:`25354` - Fails to compile when PM_DIRECT_FORCE_MODE is true
```
Comment (abrodkin) [inline doc/releases/release-notes-2.3.rst:841]: Do we really want to patch (I assume this was done automatically) docs for already done release?

**Review by abrodkin — DISMISSED** [review]
A couple of tiny things to consider.

File: arch/arc/core/reset.S (hunk 1 above)
Comment (nashif) [reply arch/arc/core/reset.S:124]: duh :)

File: doc/releases/release-notes-2.3.rst (hunk 2 above)
Comment (nashif) [reply doc/releases/release-notes-2.3.rst:841]: no, those will not be part of the change

File: subsys/power/debug.h [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+#ifdef CONFIG_PM_DEBUG
+
+struct pm_debug_info {
+	uint32_t count;
+	uint32_t last_res;
+	uint32_t total_res;
+};
+
+static struct pm_debug_info pm_dbg_info[POWER_STATE_MAX];
+static uint32_t timer_start, timer_end;
```
Comment (ceolin) [inline subsys/power/debug.h:21]: every file including this header will have its own copy of these variables. Shouldn't it be declared in C file and referenced as `extern` in the functions here ?

File: kernel/idle.c [hunk 4]
```diff
@@ -67,7 +67,7 @@ void __attribute__((weak)) pm_system_resume_from_deep_sleep(void)
  * @return N/A
  */
 #if !SMP_FALLBACK
-static void set_kernel_idle_time_in_ticks(int32_t ticks)
+static void _set_kernel_idle_time_in_ticks(int32_t ticks)
 {
```
Comment (ceolin) [inline kernel/idle.c:71]: This commit is not necessary since you get rid of this function later.

**Comment by ceolin** [comment]
looks good to me, only few comments

File: subsys/power/debug.h (hunk 3 above)
Comment (nashif) [reply subsys/power/debug.h:21]: plan is to get rid of this altogether, do not bother with that

**Review by ceolin — APPROVED** [review]
looks good for me

**Review by pabigot — APPROVED** [review]
I guess this is OK as long as we don't care about breaking anybody who's currently using the old Kconfig symbols.  That's probably fine as at least some of these are going to change again.

**Pull request merged**